tests/std/tests/P2474R2_views_repeat/test.cpp

```diff
@@ -0,0 +1,294 @@
+
+template <class T, class B>
+concept CanViewRepeat = requires(T&& w, B b) { views::repeat(forward<T>(w), b); };
+
+template <class R>
+concept CanSize = requires(R& r) { ranges::size(r); };
+
+template <class R>
+concept CanTakeDrop = requires(R r) {
+                          forward<R>(r) | views::take(1);
+                          forward<R>(r) | views::drop(1);
+                      };
+
+template <class T, class B = unreachable_sentinel_t>
+constexpr void test_common(T val, B bound = unreachable_sentinel) {
+    constexpr bool bounded = !same_as<B, unreachable_sentinel_t>;
+    using R                = ranges::repeat_view<T, B>;
+
+    static_assert(same_as<ranges::range_value_t<R>, T>);
+    static_assert(same_as<ranges::range_reference_t<R>, const T&>);
+    static_assert(same_as<ranges::range_reference_t<const R>, const T&>);
+
+    static_assert(ranges::random_access_range<R>);
+    static_assert(!ranges::contiguous_range<R>);
+    static_assert(ranges::common_range<R> == bounded);
+
+    static_assert(ranges::view<R>);
+    static_assert(semiregular<R>);
+    static_assert(default_initializable<R> == default_initializable<T>);
+    static_assert(is_nothrow_copy_constructible_v<R> == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(is_nothrow_copy_assignable_v<R> == is_nothrow_copy_assignable_v<T>); // strengthened
+    static_assert(is_nothrow_move_constructible_v<R> == is_nothrow_move_constructible_v<T>); // strengthened
+    static_assert(is_nothrow_move_assignable_v<R> == is_nothrow_move_assignable_v<T>); // strengthened
+
+    if constexpr (!bounded) {
+        static_assert(same_as<ranges::range_difference_t<R>, ptrdiff_t>);
+        static_assert(same_as<ranges::sentinel_t<R>, unreachable_sentinel_t>);
+    } else if constexpr (_Signed_integer_like<B>) {
+        static_assert(same_as<ranges::range_difference_t<R>, B>);
+    } else {
+        static_assert(same_as<ranges::range_difference_t<R>, ranges::_Iota_diff_t<B>>);
+    }
+
+    static_assert(ranges::common_range<const R> == bounded);
+    static_assert(
+        same_as<ranges::iterator_t<const ranges::repeat_view<T, B>>, ranges::iterator_t<ranges::repeat_view<T, B>>>);
+
+    static_assert(same_as<ranges::range_difference_t<const R>, ranges::range_difference_t<R>>);
+
+    static_assert(CanSize<ranges::repeat_view<T, B>> == bounded);
+
+    same_as<R> auto rng = views::repeat(val, bound);
+    static_assert(noexcept(views::repeat(val, bound)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(rng | views::drop(1)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(rng | views::take(1)) == is_nothrow_copy_constructible_v<T>); // strengthened
+    static_assert(noexcept(std::move(rng) | views::drop(1)) == is_nothrow_move_constructible_v<T>); // strengthened
+    static_assert(noexcept(std::move(rng) | views::take(1)) == is_nothrow_move_constructible_v<T>); // strengthened

```

Comment:
Qualifying `std::move` is no longer necessary now that we've silenced the Clang warning.

Reply:
Heh. I add a commit removing them, then I go ahead and add some new ones...